src/index.ts

```diff
@@ -39,30 +41,38 @@ const parseEnvironmentVariables = async () => {
   };
 };
 
-const getPullRequestsToPublish = async (
-  githubToken: string,
-  repositories: ReadonlyArray<string>
-) => {
-  const {
-    data: { items: pullRequests },
-  } = await new Octokit({
-    auth: githubToken,
-  }).rest.search.issuesAndPullRequests({
-    q: `state:open type:pr author:app/dependabot ${repositories
-      .map((repo) => `repo:${repo}`)
-      .join(" ")}`,
-  });
-
-  return pullRequests.map((pr) => ({
-    title: pr.title,
-    url: pr.html_url,
-    repo: pr.repository_url.replace("https://api.github.com/repos/inato/", ""),
-  }));
-};
+const getPullRequestsToPublish = () =>
+  pipe(
+    reader.asks(
+      ({
+          githubToken,
+          repositories,
+        }: {
+          githubToken: string;
+          repositories: ReadonlyArray<string>;
+        }) =>
+        () =>
+          new Octokit({
+            auth: githubToken,
+          }).rest.search.issuesAndPullRequests({
+            q: `state:open type:pr author:app/dependabot ${repositories
+              .map((repo) => `repo:${repo}`)
+              .join(" ")}`,
+          })
+    ),
+    readerTask.map(({ data: { items } }) =>
+      items.map((pullRequest) => ({
+        title: pullRequest.title,
+        url: pullRequest.html_url,
+        repo: pullRequest.repository_url.replace(
+          "https://api.github.com/repos/inato/",
```

Comment:
This can be extracted in a named variable

Reply:
I even extracted the logic in a function to have it clearer. Thanks